pkg/queue/queue.go

```diff
@@ -71,3 +74,11 @@ func (q *Queue) AddIfNotPresent(w *workload.Info) bool {
 	}
 	return false
 }
+
+func (q *Queue) reportPendingWorkloads() {
+	metrics.PendingWorkloads.WithLabelValues(q.ClusterQueue, q.Key).Set(float64(len(q.items)))
```

Comment:
why not using `Key(q)` directly but introduced a new field `Key` here?

Reply:
Maybe can't. 
We don't have the object `*kueue.Queue` in the struct `Queue`.